### DISCUSSION TITLE
Is the logging lib supposed to be stack based?

### DISCUSSION THREAD

**Discussion by aadeg** [eclipse-iceoryx/iceoryx#1040] [Q&A] [answered]
Looking at the source code of the logging module, I discover that it uses the heap. For example the `LogEntry` struct is defined as:
```cpp
struct LogEntry
{
    LogLevel level{LogLevel::kVerbose};
    std::chrono::milliseconds time{0};
    std::string message;   //  <------- heap
};
```

Isn't it against the rules written in the CONTRIBUTION.md?

> 1. No heap is allowed, static memory management hugely decreases the complexity of your software (e.g. cxx::vector without heap)

**Comment by elBoberido** [comment, marked as answer]
@aadeg this was just a stop gap solution. I'm currently working on a design document to overhaul the logger and also remove the heap usage. There are a few other places where currently a `std::string` is [used](https://github.com/eclipse-iceoryx/iceoryx/issues/260#issuecomment-731246365). All of this will be fixed in the future.

**Comment by aadeg** [comment]
Great news! Thanks for the clarification
